File: DAL/Models/Employee.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace DAL.Models
{
    public class Employee
    {
        public int Id { get; set; }

        public string UserId { get; set; }
        
        public DateTime BirthDate { get; set; }

        public Enums.Gender Gender { get; set; }
        public Enums.MilitaryStatus MilitaryStatus { get; set; }

        public City City { get; set; }
        public int? CityId { get; set; }
        public int? CountryId { get; set; }
        public Country Country { get; set; }
        public bool IsWillingToRelocate { get; set; }
        public string MobileNumber { get; set; }
        public string AlternativeMobileNumber { get; set; }
        public int? CareerLevelId { get; set; }
        public CareerLevel CareerLevel { get; set; }
        public ICollection<JobType> JobTypes { get; set; }
        public ICollection<JobCategory> PreferredJobCategories { get; set; }
        public int MinimumSalary { get; set; }
        public int ExperienceYears { get; set; }
        public ICollection<Skill> Skills { get; set; }
        public ICollection<UserLanguage> UserLanguages { get; set; }
        public ICollection<EmployeeLinks> EmployeeLinks { get; set; }
        public string CV { get; set; }
        public string Photo { get; set; }
        public string Summary { get; set; }
        public int? EducationLevelId { get; set; }
        public EducationLevel EducationLevel { get; set; }
        public int? NationalityId { get; set; }
        public Country Nationality { get; set; }

        public ApplicationUser User { get; set; }
        public ICollection<Application> Applications { get; set; }
    }
}

File: GlassDoor/ClientApp/src/app/models/employee.ts
import { Application } from "./application";
import { ApplicationUser } from "./application-user";
import { CareerLevel } from "./career-level";
import { City } from "./city";
import { Country } from "./country";
import { EducationLevel } from "./education-level";
import { EmployeeLinks } from "./employee-links";
import { Gender } from "./enums/gender.enum";
import { MilitaryStatus } from "./enums/military-status.enum";
import { JobCategory } from "./job-category";
import { JobTypes } from "./job-types";
import { Skills } from "./skills";
import { UserLanguage } from "./user-language";

export class Employee {


  constructor(public id?: number, public userId?: string, public birthDate?: Date, public gender?: Gender,
    public militaryStatus?: MilitaryStatus, public cityName?: string,public countryId?: number,
    public countryName?: string, public isWillingToRelocate?: boolean,
    public mobileNumber?: string, public alternativeMobileNumber?: string, public careerLevelId?: number,
    public careerLevelName?: string, public jobTypesName?: JobTypes[], public preferredJobCategoriesName?: JobCategory,
    public minimumSalary?: number, public experienceYears?: number, public skillsNames?: Skills[],
    public userLanguagesNames?: UserLanguage[], public employeeLinksNames?: EmployeeLinks[],public cV?: string,
    public photo?: string, public summary?: string, public educationLevelId?: number,
    public educationLevelName?: string, public nationalityId?: number, public nationalityName?: string,
    public userFirstName?: string, public userLastName?: string, public email?: string)
  {
  }

}
